jetbrains-core/src/software/aws/toolkits/jetbrains/services/ecs/exec/RunCommandRunProfile.kt

```diff
@@ -0,0 +1,127 @@
+// Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.services.ecs.exec
+
+import com.intellij.execution.ExecutionResult
+import com.intellij.execution.Executor
+import com.intellij.execution.configurations.CommandLineState
+import com.intellij.execution.configurations.GeneralCommandLine
+import com.intellij.execution.configurations.PtyCommandLine
+import com.intellij.execution.configurations.RunProfile
+import com.intellij.execution.configurations.RunProfileState
+import com.intellij.execution.filters.RegexpFilter
+import com.intellij.execution.filters.TextConsoleBuilderFactory
+import com.intellij.execution.process.ColoredProcessHandler
+import com.intellij.execution.process.ProcessAdapter
+import com.intellij.execution.process.ProcessEvent
+import com.intellij.execution.process.ProcessHandler
+import com.intellij.execution.process.ProcessOutputTypes
+import com.intellij.execution.process.ProcessTerminatedListener
+import com.intellij.execution.runners.ExecutionEnvironment
+import com.intellij.execution.runners.ProgramRunner
+import com.intellij.execution.ui.RunContentManager
+import com.intellij.ide.macro.Macro
+import com.intellij.ide.macro.MacroManager
+import com.intellij.openapi.actionSystem.DataContext
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.util.Key
+import com.intellij.openapi.util.registry.Registry
+import com.intellij.openapi.util.text.StringUtil
+import com.intellij.tools.FilterInfo
+import com.intellij.tools.Tool
+import com.intellij.tools.ToolProcessAdapter
+import com.intellij.tools.ToolRunProfile
+import com.intellij.tools.ToolsCustomizer
+import java.io.File
+import javax.swing.Icon
+
+class RunCommandRunProfile(private val tool: Tool, private val dataContext: DataContext, private val credentials: Map<String, String>) : RunProfile {
```

Comment:
Do we still need the tool? Or should we just take in a GenralCommandLine?

Reply:
made the changes to remove tool and just use GeneralCommandLine